klient/conf/config.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conf
+
+import (
+	"flag"
+	"os"
+	"os/user"
+	"path"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var DefaultClusterContext = ""
+
+// New returns Kubernetes configuration value of type *rest.Config.
+func New(fileName string) (*rest.Config, error) {
+	// if filename is not provided assume in-cluster-config
+	if fileName == "" {
+		return rest.InClusterConfig()
+	}
+
+	// create the config object from k8s config path
+	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: fileName}, &clientcmd.ConfigOverrides{}).ClientConfig()
+}
+
+// NewWithContextName returns k8s config value of type *rest.Config
+func NewWithContextName(fileName, context string) (*rest.Config, error) {
+	// create the config object from k8s config path and context
+	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: fileName},
+		&clientcmd.ConfigOverrides{
+			CurrentContext: context,
+		}).ClientConfig()
+}
+
+// ResolveKubeConfigFile returns the kubeconfig file from
+// either flag --kubeconfig or env KUBECONFIG.
+// It will only call flag.Parse() if flag.Parsed() is false.
```

Comment:
This comment is wrong, the code does not call `flag.Parse()` directly and shouldn't.  As you have in the code, it detects to see if flag.Parsed() is true, then look for the proper flag.